### DISCUSSION TITLE
Remote generation Windows + Mac OS

### DISCUSSION THREAD

**Discussion by quasar136a8** [lllyasviel/Fooocus#2539] [General]
A question has. I have a desktop computer on Windows with 4090 cards, and a MacBook on OS.
Is it possible to somehow generate photos on a MacBook using the power of a Windows computer and a 4090 graphics card?
To use only the ability of a PC without a remote connection via Teamviewer, for example?

**Comment by mashb1t** [comment]
You can add `--listen` after entry_with_update.py to the run.bat of the Windows PC, then access Fooocus using the IP of the WIndows PC in the browser, e.g. 192.168.178.123:7865

**Reply by quasar136a8** [reply]
Like this:

.\python_embeded\python.exe -s Fooocus\entry_with_update.py --listen --preset realistic
pause
--- After ---
App started successfully. Use the app with http://localhost:7865/ or 0.0.0.0:7865

Not work if I write this address on a mac

![CleanShot 2024-03-16 at 01 35 49@2x](https://github.com/lllyasviel/Fooocus/assets/111929719/e98e53aa-2695-4d08-b2b5-4c01fe6a9949)

**Reply by mashb1t** [reply]
Yes like this, but you have to use the actual IP of your Windows PC in your Macbooks browser address bar, which you can find in your network settings.
FYI 0.0.0.0 is basically the opposite of 127.0.0.1, also a synthetic IP. The output shows that you can connect from your machine as well as any other device in your network.

**Reply by quasar136a8** [reply]
yes, I can open  interface  but in styles nothing, screen
![CleanShot 2024-03-16 at 01 49 43@2x](https://github.com/lllyasviel/Fooocus/assets/111929719/cc7effbc-f0e1-4d7b-bece-83da22187d1b)

**Reply by quasar136a8** [reply]
It opening but nothing happening when I press "Generate"
![CleanShot 2024-03-16 at 01 52 28@2x](https://github.com/lllyasviel/Fooocus/assets/111929719/51ae36a8-fb2e-441d-a766-20c959be9b09)

**Reply by mashb1t** [reply]
please check the terminal output on the windows machine, your screenshot shows 3 in queue.

**Reply by quasar136a8** [reply]
Thank you so much all works well work now
![CleanShot 2024-03-16 at 02 00 26@2x](https://github.com/lllyasviel/Fooocus/assets/111929719/094b895e-5c4b-437d-b155-2c3508212e72)

**Reply by quasar136a8** [reply]
Another problem.
Generation works well if not use an Image prompt. If I need to use an Image prompt. I have an error
<img width="830" alt="Screenshot 2024-03-16 145740" src="https://github.com/lllyasviel/Fooocus/assets/111929719/714bbb20-a8f1-42e5-8cfb-f024c0e0da71">
Its not work with the Image prompt.

**Reply by mashb1t** [reply]
already fixed in https://github.com/lllyasviel/Fooocus/pull/2537, new release 2.3.0-rc1 will be out today

**Reply by quasar136a8** [reply]
Thank you!
